frameworks/spark/src/test/java/com/criteo/hadoop/garmadon/spark/listener/GarmadonSparkStorageStatusListenerIntegrationTest.java

```diff
@@ -0,0 +1,215 @@
+        assertTrue(executors.isEmpty());
+
+        //Memory
+        JavaRDD rddMemory = makeRDD("MemRDD", StorageLevel.MEMORY_ONLY());
+        rddMemory.collect();
+
+        checkExecutorRDDStorage("driver", equalTo(0L), greaterThan(0L), equalTo(0L));
+
+        //Disk
+        JavaRDD rddDisk = makeRDD("DiskRDD", StorageLevel.DISK_ONLY());
+        rddDisk.collect();
+
+        checkExecutorRDDStorage("driver", equalTo(0L), greaterThan(0L), greaterThan(0L));
+
+        //OffHeap
+        JavaRDD rddOffHeap = makeRDD("OffHeapRDD", StorageLevel.OFF_HEAP());
+        rddOffHeap.collect();
+
+        checkExecutorRDDStorage("driver", greaterThan(0L), greaterThan(0L), greaterThan(0L));
+
+        rddMemory.unpersist(true);
+        //wait for the EventBus to fire the unpersistRDD event
+        Thread.sleep(1000);
+        checkExecutorRDDStorage("driver", greaterThan(0L), equalTo(0L), greaterThan(0L));
+
+        rddDisk.unpersist(true);
+        //wait for the EventBus to fire the unpersistRDD event
+        Thread.sleep(1000);
+        checkExecutorRDDStorage("driver", greaterThan(0L), equalTo(0L), equalTo(0L));
+
+        rddOffHeap.unpersist(true);
+        //wait for the EventBus to fire the unpersistRDD event
+        Thread.sleep(1000);
+        checkExecutorRDDStorage("driver", equalTo(0L), equalTo(0L), equalTo(0L));
+    }
+
+    @Test
+    public void SparkStorageStatusListener_should_track_broadcast_storage_status() throws InterruptedException {
+        assertTrue(rdds.isEmpty());
+        assertTrue(executors.isEmpty());
+
+        Broadcast<List<UUID>> broadcast = jsc.broadcast(IntStream.range(0, 1000000).mapToObj(i -> UUID.randomUUID()).collect(Collectors.toList()));
+
+        jsc.parallelize(Collections.singletonList(1)).map(i -> broadcast.getValue()).collect();
+
+        checkExecutorBroadcastStorage("driver", equalTo(0L), greaterThan(0L), equalTo(0L));
+    }
+
+    private void checkRddStorage(String name, Matcher<Long> offHeapMatcher, Matcher<Long> memoryMatcher, Matcher<Long> diskMatcher) {
+        RDDInfo info = rdds.get(name);
+        assertThat("offheap storage has unexpected value", info.offHeap, offHeapMatcher);
+        assertThat("memory storage has unexpected value", info.memory, memoryMatcher);
+        assertThat("disk storage has unexpected value", info.disk, diskMatcher);
+    }
+
+    private void checkExecutorRDDStorage(String name, Matcher<Long> offHeapMatcher, Matcher<Long> memoryMatcher, Matcher<Long> diskMatcher) {
+        ExecutorInfo info = executors.get(name);
+        assertThat("rdd offheap storage has unexpected value", info.rddOffHeap, offHeapMatcher);
+        assertThat("rdd memory storage has unexpected value", info.rddMemory, memoryMatcher);
+        assertThat("rdd disk storage has unexpected value", info.rddDisk, diskMatcher);
+    }
+
+    private void checkExecutorBroadcastStorage(String name, Matcher<Long> offHeapMatcher, Matcher<Long> memoryMatcher, Matcher<Long> diskMatcher) {
```

Comment:
You wrote this method as if you wanted to check offHeap and disk, but you never do.

Reply:
true, I don't really know if broadcast can go to disk or off heap and how it could be triggered. So, I just test that it stays to 0.
Do you want me to change something or was that to question if I could add such tests (on disk/offheap) or not ?

Reply:
My point is this is a bit overkill when none of your matcher ever changes value. I also wanted to check that you didn't unintentionally forget test cases. But this makes the code more explicit, so I'm still OK with it.